utils/browser.py

```diff
@@ -63,21 +63,24 @@ def ensure_browser_open():
     """
     try:
         browser().current_url
+        if item and browser().name not in str(item).lower():
+            browser().quit()
+            start(item=item)
     except UnexpectedAlertPresentException:
         # Try to handle an open alert, restart the browser if possible
         try:
             browser().switch_to_alert().dismiss()
         except:
-            start()
+            start(item=item)
     except:
         # If we couldn't poke the browser for any other reason, start a new one
-        start()
+        start(item=item)
     if thread_locals.wharf:
         thread_locals.wharf.renew()
     return browser()
 
 
-def start(webdriver_name=None, base_url=None, **kwargs):
+def start(item=None, webdriver_name=None, base_url=None, **kwargs):
```

Comment:
Instead of us passing ```item``` in here, the better approach is to get ```ensure_browser_open``` to runtime override the conf, so that ```start``` can just use the value from the conf module.